### PR TITLE
Add integration tests(1)

### PR DESCRIPTION
**Pull request by duguorong009**
Summary of changes
Changes introduced in this pull request
   - Implement the integration tests using CosmJS + (local) juno
   - Add the integration tests for `TokenWrapper`, `mixer` & `anchor` contracts
    
Reference issue to close (if applicable)
Closes

### CODE REVIEW

**Review by shekohex — CHANGES_REQUESTED** [review]
I will take another review, but this PR needs a lot of cleanups

remove `test-scripts/.DS_Store` and all `.DS_Store` files.
and also, would be could to remove all `*.wasm` files, this could be generated/compiled later during the tests.

**Comment by duguorong009** [comment]
> I will take another review, but this PR needs a lot of cleanups
> 
> remove `test-scripts/.DS_Store` and all `.DS_Store` files. and also, would be could to remove all `*.wasm` files, this could be generated/compiled later during the tests.

Done https://github.com/webb-tools/protocol-cosmwasm/pull/53/commits/7c37a80e029b07c53071a17b7f4798f6b3fddde9

File: test-scripts/src/environments/testnet.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    junod = await SigningCosmWasmClient.connectWithSigner(config.networkInfo.url, wallet1);
+
+    const [account1] = await wallet1.getAccounts();
+    const [account2] = await wallet2.getAccounts();
+    const [account3] = await wallet3.getAccounts();
+
+    console.log(`Use ${chalk.cyan(account1.address)} as Wallet 1`);
+    console.log(`Use ${chalk.cyan(account2.address)} as Wallet 2`);
+    console.log(`Use ${chalk.cyan(account3.address)} as Wallet 3`);
+    // console.log(`Use ${chalk.cyan(wallet4.key.accAddress)} as Wallet 4`);
```
Comment (dutterbutter) [inline test-scripts/src/environments/testnet.ts:52]: Please remove commented console log

File: test-scripts/src/index.ts [hunk 2]
```diff
@@ -1,23 +1,34 @@
...
+			await LocalNet.startTests();
+			break;
+		case "localjuno_setup_contracts":
+			await LocalNet.startSetupContracts();
+			break;
+		case "localjuno_migrate_contracts":
+			// await LocalNet.startMigrateContracts();
+			break;
+
+
```
Comment (dutterbutter) [inline test-scripts/src/index.ts:20]: Remove spacing

File: test-scripts/src/index.ts [hunk 3]
```diff
@@ -1,23 +1,34 @@
...
+	const mode = process.env.npm_config_mode || "";
+	switch (mode) {
+		case "localjuno_tests":
+			await LocalNet.startTests();
+			break;
+		case "localjuno_setup_contracts":
+			await LocalNet.startSetupContracts();
+			break;
+		case "localjuno_migrate_contracts":
+			// await LocalNet.startMigrateContracts();
```
Comment (dutterbutter) [inline test-scripts/src/index.ts:17]: Remove commented out statement

File: test-scripts/src/processes/tests/anchor.ts [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+    ).to.be.rejected; // rejectedWith("Commitment not found");
+  
+    // Succeed to "deposit"
+    const result = await  wallet3_client.execute(localjuno.addresses.wallet3, localjuno.contracts.anchor, {
+      wrap_and_deposit: {
+        commitment: [235, 3, 87, 22, 66, 107, 162, 89, 147, 182, 207, 54, 195, 138, 136, 110, 150, 128, 213, 43, 44, 121, 177, 9, 17, 54, 9, 202, 11, 230, 158, 8],
+        amount: ucosm_amount,
+      }
+    }, "auto", undefined, [coin(ucosm_to_send, "ucosm")])
+    // console.log(result);
```
Comment (dutterbutter) [inline test-scripts/src/processes/tests/anchor.ts:87]: Please remove commented out code.

File: test-scripts/src/processes/tests/testnet.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  // await testAnchorUnwrapNative(junod, anchor, wallet3, "100");
+  // await testAnchorWrapCw20(junod, anchor, tokenWrapper, cw20, wallet3, "10000");
+  // await testAnchorUnwrapCw20(junod, anchor, tokenWrapper, cw20, wallet3, "100");
+
+  // VAnchor
+
+  // // Mixer
+  // await testMixerInitialize(junod, mixer);
+  // await testMixerDepositNativeToken(junod, mixer, wallet3, "1000000");
+  // await testMixerWithdrawNativeToken(junod, mixer, wallet1, wallet2, wallet3, "1000000");
```
Comment (dutterbutter) [inline test-scripts/src/processes/tests/testnet.ts:69]: Can we remove commented out code?

File: test-scripts/src/processes/tests/tokenWrapper.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+// ------------------------------------------------
+export async function testTokenWrapperInitialize(
+    junod: SigningCosmWasmClient,
+    tokenWrapper: string,
+  ): Promise<void> {
+    process.stdout.write("Test - TokenWrapper should initialize");
+    const result: any = await junod.queryContractSmart(tokenWrapper, {
+      config: {},
+    });
+    // console.log(result);
```
Comment (dutterbutter) [inline test-scripts/src/processes/tests/tokenWrapper.ts:29]: Remove commented out console log

File: test-scripts/src/utils/helpers.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+/**
+ * @notice Upload contract code to LocalJuno. Return code ID.
+ */
+export async function storeCode(
+  junod: SigningCosmWasmClient,
+  deployer: DirectSecp256k1HdWallet,
+  filepath: string
+): Promise<number> {
+  const code = fs.readFileSync(filepath);
+  // const [account] = await deployer.getAccounts();
```
Comment (dutterbutter) [inline test-scripts/src/utils/helpers.ts:34]: Remove commented out code

File: test-scripts/src/utils/helpers.ts [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export async function instantiateContract(
+  junod: SigningCosmWasmClient,
+  deployer: DirectSecp256k1HdWallet,
+  admin: DirectSecp256k1HdWallet, // leave this emtpy then contract is not migratable
+  codeId: number,
+  instantiateMsg: Record<string, unknown>
+) {
+  // const [account] = await deployer.getAccounts();
```
Comment (dutterbutter) [inline test-scripts/src/utils/helpers.ts:51]: Remove commented out code

**Review by dutterbutter — CHANGES_REQUESTED** [review]
Please add documentation to public structs / enums / functions. Please remove all commented out code. Great start!

File: test-scripts/src/utils/helpers.ts [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export async function migrateContract(
+  junod: SigningCosmWasmClient,
+  sender: DirectSecp256k1HdWallet,
+  admin: DirectSecp256k1HdWallet,
+  contract: string,
+  new_code_id: number,
+  migrateMsg: Record<string, unknown>
+) {
+  // const [account] = await sender.getAccounts();
```
[the remark this answers is not in the document]
Comment (duguorong009) [reply test-scripts/src/utils/helpers.ts]: Done https://github.com/webb-tools/protocol-cosmwasm/pull/53/commits/8bc0c23bd2f0ea698680ecb5153284f136c0a0cc

File: test-scripts/src/index.ts (hunk 3 above)
Comment (duguorong009) [reply test-scripts/src/index.ts]: Done https://github.com/webb-tools/protocol-cosmwasm/pull/53/commits/8bc0c23bd2f0ea698680ecb5153284f136c0a0cc

File: test-scripts/src/processes/tests/tokenWrapper.ts (hunk 6 above)
Comment (duguorong009) [reply test-scripts/src/processes/tests/tokenWrapper.ts]: Done https://github.com/webb-tools/protocol-cosmwasm/pull/53/commits/8bc0c23bd2f0ea698680ecb5153284f136c0a0cc

File: test-scripts/src/utils/helpers.ts (hunk 7 above)
Comment (duguorong009) [reply test-scripts/src/utils/helpers.ts]: Done https://github.com/webb-tools/protocol-cosmwasm/pull/53/commits/8bc0c23bd2f0ea698680ecb5153284f136c0a0cc

**Review by dutterbutter — APPROVED** [review]

**Review by shekohex — APPROVED** [review]

**Pull request merged**